actionpack/lib/action_controller/metal/head.rb

```diff
@@ -17,8 +17,14 @@ module Head
     #
     # See Rack::Utils::SYMBOL_TO_STATUS_CODE for a full list of valid +status+ symbols.
     def head(status, options = {})
-      options, status = status, nil if status.is_a?(Hash)
-      status ||= options.delete(:status) || :ok
+      if status.is_a?(Hash)
+        msg = "Passing Hash as first parameter"
+        options, status = status, nil
+        status ||= options.delete(:status) || (
+          (msg += " and default status code") && :ok
+        )
+        ActiveSupport::Deprecation.warn(msg + " for `head` is deprecated and will be removed in Rails 5.1. Please visit http://api.rubyonrails.org/classes/ActionController/Head.html for usage.")
```

Comment:
I just don't remember us doing that anywhere else. Besides they're calling head and you even mention it in the warning, so I'd rather :scissors: it.